etcdserver/server.go

```diff
@@ -247,7 +247,7 @@ func NewServer(cfg *ServerConfig) (*EtcdServer, error) {
 			var err error
 			str, err = discovery.JoinCluster(cfg.DiscoveryURL, cfg.DiscoveryProxy, m.ID, cfg.InitialPeerURLsMap.String())
 			if err != nil {
-				return nil, err
+				return nil, fmt.Errorf("error joining discovery cluster (%v)", err)
```

Comment:
I want to display the exact reason that joining cluster failed, so people could dig into it. Do you mention to define the constant part `error joining discovery cluster` as an error string?